src/hooks/useViewport/ViewportProvider.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { createContext } from 'react';
+import { useViewport, IUseViewportOptions } from './useViewport';
+
+interface IViewportSize {
+  width: number;
+  height: number;
+}
+
+type TViewportContext = IViewportSize | undefined;
+
+const ViewportContext = createContext<TViewportContext>(undefined);
```

Comment:
I think it will be better to just pass `undefined` as default to `width` and `height` the 
```suggestion
const ViewportContext = createContext({
  width: undefined,
  height: undefined, 
});
```
then we can rid of the `interface IViewportSize` of `type TViewportContext`